src/renderer/gdi/state.cpp

```diff
@@ -438,8 +439,12 @@ GdiEngine::~GdiEngine()
     // Now fill up the FontInfo we were passed with the full details of which font we actually chose
     {
         // Get the actual font face that we chose
-        WCHAR wszFaceName[LF_FACESIZE];
-        RETURN_HR_IF(E_FAIL, !(GetTextFaceW(hdcTemp.get(), ARRAYSIZE(wszFaceName), wszFaceName)));
+        const auto faceNameLength{ GetTextFaceW(hdcTemp.get(), 0, nullptr) };
+
+        wistd::unique_ptr<wchar_t[]> currentFaceName{ wil::make_unique_nothrow<wchar_t[]>(faceNameLength) };

```

Comment:
I think the preferred pattern now is to use a `std::wstring` and `.resize()` it to the size that you will need to fit and fill it instead of using a smart pointer. I'm pretty sure Audit Mode throws a warning on using a smart pointer (instead of some sort of collection or string) for things that only really span the lifetime of a single function body.